### PR TITLE
fetch time bin from conf and updated missing xaxis.

### PR DESCRIPTION
**Pull request by rajnikantsh**
hardcode value of time bin replaced from beam.conf timebin value.

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/508)

### CODE REVIEW

File: src/main/java/beam/analysis/plots/GraphsStatsAgentSimEventsListener.java [hunk 1]
```diff
@@ -42,16 +42,17 @@
     private IGraphStats fuelUsageStats = new FuelUsageStats(new FuelUsageStats.FuelUsageStatsComputation());
     private IGraphStats modeChoseStats = new ModeChosenStats(new ModeChosenStats.ModeChosenComputation());
     private IGraphStats personTravelTimeStats = new PersonTravelTimeStats(new PersonTravelTimeStats.PersonTravelTimeComputation());
-    private IGraphStats personVehicleTransitionStats = new PersonVehicleTransitionStats();
     private IGraphStats rideHailWaitingStats;
     //private IGraphStats generalStats = new RideHailStats();
+    private IGraphStats personVehicleTransitionStats;
```
Comment (carloscaldas) [inline src/main/java/beam/analysis/plots/GraphsStatsAgentSimEventsListener.java]: Please make this field a blank final variable as follows:
```Java
private final IGraphStats personVehicleTransitionStats;
```

File: src/main/java/beam/analysis/plots/PersonVehicleTransitionStats.java [hunk 2]
```diff
@@ -32,8 +33,13 @@
     private static Map<String, TreeMap<Integer, Integer>> onRoutes = new HashMap();
     private static Map<String, Integer> modePerson = new HashMap<>();
     private static final String fileName = "tripHistogram";
-    private final int binSize = 300;
-    private final int nofBins = 30 * 3600 / binSize + 1;
+    private  int binSize = 0;
+    private  int nofBins = 0;
```
Comment (carloscaldas) [inline src/main/java/beam/analysis/plots/PersonVehicleTransitionStats.java]: Suggestions:
- Fix the formatting: there are two spaces between `private` and `int`. 
- There is no need to initialize with default value.
- Make the name of the variable `nofBins` more meaningful.

```Java
private int binSize;
private int numOfBins;
```

File: src/main/java/beam/analysis/plots/PersonVehicleTransitionStats.java [hunk 3]
```diff
@@ -32,8 +33,13 @@
...
+
+    PersonVehicleTransitionStats(BeamConfig beamConfig){
+        binSize = beamConfig.beam().outputs().stats().binSize();
+        nofBins = 30 * 3600 / binSize + 1;
```
Comment (carloscaldas) [inline src/main/java/beam/analysis/plots/PersonVehicleTransitionStats.java]: Consider replacing the magic numbers `30 * 3600` with a meaningful constant. Where those numbers came from? `3600` looks like the number of seconds in one hour and `30` looks like the number of days in a month. It is hard to infer what is the purpose of this combination.

File: src/main/java/beam/analysis/plots/PersonVehicleTransitionStats.java [hunk 4]
```diff
@@ -223,7 +229,7 @@ JFreeChart getGraphic(String mode, int iteration) {
         plot.getRangeAxis().setStandardTickUnits(NumberAxis.createIntegerTickUnits());
         final CategoryAxis axis1 = new CategoryAxis("sec");
         axis1.setTickLabelFont(new Font("SansSerif", Font.PLAIN, 7));
-        plot.setDomainAxis(new NumberAxis("time"));
+        plot.setDomainAxis(new NumberAxis("timeBinSize=3600 sec"));
```
Comment (carloscaldas) [inline src/main/java/beam/analysis/plots/PersonVehicleTransitionStats.java]: The _magic string_ `timeBinSize=3600 sec` is repeated two times in this class. Consider extracting a constant with a meaningful name.

Comment (carloscaldas) [inline src/main/java/beam/analysis/plots/PersonVehicleTransitionStats.java]: The _magic string_ `timeBinSize=3600 sec` is repeated two times in this class. Consider extracting a constant with a meaningful name.

File: src/main/java/beam/analysis/plots/GraphsStatsAgentSimEventsListener.java [hunk 5]
```diff
@@ -42,16 +42,17 @@
     private IGraphStats fuelUsageStats = new FuelUsageStats(new FuelUsageStats.FuelUsageStatsComputation());
     private IGraphStats modeChoseStats = new ModeChosenStats(new ModeChosenStats.ModeChosenComputation());
     private IGraphStats personTravelTimeStats = new PersonTravelTimeStats(new PersonTravelTimeStats.PersonTravelTimeComputation());
-    private IGraphStats personVehicleTransitionStats = new PersonVehicleTransitionStats();
     private IGraphStats rideHailWaitingStats;
     //private IGraphStats generalStats = new RideHailStats();
```
Comment (carloscaldas) [inline src/main/java/beam/analysis/plots/GraphsStatsAgentSimEventsListener.java]: Since you refactored the code, consider removing commited comments.

**Comment by rajnikantsh** [comment]
The changes of this branch were moved to 4ci branch

**Pull request closed without merging**